packages/web/hooks/use-swap.tsx

```diff
@@ -408,6 +408,17 @@ export function useSwap(
             );
           }
 
+          // Note about One-Click Trading (1CT) synchronization:
+          //
+          // When 1CT is enabled via the swap modal, there's a timing mismatch between
+          // the local state (isOneClickTradingEnabled) and the account store state.
+          //
+          // As a result:
+          // 1. The first swap after enabling 1CT will require manual signing
+          // 2. Subsequent swaps will use 1CT automatically
+          //
+          // This occurs because 1CT transactions require more gas, which makes
+          // our current quote's gas estimation incorrect and low. This results in out of gas error.
```

Comment:
I wonder if, when enabling the 1CT session in the swap tool, we could wait for the 1CT session to become active before opening a second transaction modal

Reply:
based on the fact that both of you have asked for this I will assume that the changes in the gas fee is not a problem 

the naive approach of just waiting a bit did not work for me previously, but I will give it a more through check

Reply:
Ok, so solving this ended up being a bigger thing than I hoped it to be, but it was lots of fun to tackle :) 

I have a simple wait solution in 70258a9060a0 but that solution felt uncomfortably brittle so I ended up reworking the mobx <> 1CT session synchronization in 2758eef306bc and 747dbc3c406c